### PR TITLE
[Improve][e2e] Container only copy required connector jars

### LINKED ISSUE #2681: [Improve][e2e] container only copy required connector jars
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Copy the connector dependencies to the container during the e2e test: this will copy all the connectors of the entire dist, which is unnecessary. I am going to use plugin.mapping and conf parsing to only copy the required connectors;

### Usage Scenario

e2e

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
1. Container only copy required connector jars.
2. Improve code reuse

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by ashulin** [comment]
PTAL, thanks. @ruanwenjun @hailin0 @laglangyue

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/ContainerUtil.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        final String targetConfInContainer = Paths.get("/tmp", confFile).toString();
+        container.copyFileToContainer(MountableFile.forHostPath(getConfigFile(confFile).getAbsolutePath()), targetConfInContainer);
+        return targetConfInContainer;
+    }
+
+    public static void copySeaTunnelStarter(GenericContainer<?> container,
+                                              String startModuleName,
+                                              String startModulePath,
+                                              String seatunnelHomeInContainer,
+                                              String startShellName) {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/ContainerUtil.java]: align indent

File: seatunnel-e2e/seatunnel-flink-e2e-common/src/test/java/org/apache/seatunnel/e2e/flink/AbstractFlinkContainer.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    protected final String startModuleName;
+
+    protected final String startModulePath;
+
+    protected final String connectorsRootPath;
+
+    protected final String connectorType;
+
+    protected final String connectorNamePrefix;
+    protected static final Network NETWORK = Network.newNetwork();
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-e2e-common/src/test/java/org/apache/seatunnel/e2e/flink/AbstractFlinkContainer.java]: move static to up

File: seatunnel-e2e/seatunnel-flink-e2e-common/src/test/java/org/apache/seatunnel/e2e/flink/AbstractFlinkContainer.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        this.connectorsRootPath = connectorsRootPath;
+        this.connectorType = connectorType;
+        this.connectorNamePrefix = connectorNamePrefix;
+        String[] moudules = startModuleNameInSeaTunnelCore.split(File.separator);
+        this.startModuleName = moudules[moudules.length - 1];
+        this.startModulePath = PROJECT_ROOT_PATH + File.separator +
+            START_ROOT_MODULE_NAME + File.separator + startModuleNameInSeaTunnelCore;
+    }
+
+    private static final String FLINK_PROPERTIES = String.join(
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-e2e-common/src/test/java/org/apache/seatunnel/e2e/flink/AbstractFlinkContainer.java]: move static to up

File: seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java [hunk 4]
```diff
@@ -17,128 +17,49 @@
...
-            MountableFile.forHostPath(seatunnelCoreFlinkJarPath),
-            FLINK_JAR_PATH);
-
-        // copy bin
-        String seatunnelFlinkBinPath = PROJECT_ROOT_PATH + "/seatunnel-core/seatunnel-core-flink-sql/src/main/bin/start-seatunnel-sql.sh";
-        jobManager.copyFileToContainer(
-            MountableFile.forHostPath(seatunnelFlinkBinPath),
-            Paths.get(SEATUNNEL_BIN).toString());
+    @Override
+    public Container.ExecResult executeSeaTunnelFlinkJob(String confFile) throws IOException, InterruptedException {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java]: No need to call `copyConnectorJarToContainer`?

Comment (ashulin) [reply seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java]: The old `flink-sql-starter` don’t call copyConnectorJarToContainer.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java]: 👌

**Comment by hailin0** [comment]
LGTM

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

File: seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java (hunk 4 above)
Comment (ashulin) [reply seatunnel-e2e/seatunnel-flink-sql-e2e/setunnel-connector-flink-sql-e2e-base/src/test/java/org/apache/seatunnel/e2e/flink/sql/FlinkContainer.java]: Without this logic is really wrong, I've added `copyConnectorJarToContainer`.
There are other problems with flink-sql-e2e. At present, it only uses the built-in connector of flink, so there is no error. I will solve this problem in other PRs.

**Pull request merged**